junit/src/main/java/net/neoforged/neoforge/junit/utils/EphemeralTestServerProvider.java

```diff
@@ -0,0 +1,307 @@
+import net.minecraft.world.level.GameType;
+import net.minecraft.world.level.LevelSettings;
+import net.minecraft.world.level.WorldDataConfiguration;
+import net.minecraft.world.level.dimension.LevelStem;
+import net.minecraft.world.level.levelgen.WorldDimensions;
+import net.minecraft.world.level.levelgen.WorldOptions;
+import net.minecraft.world.level.levelgen.presets.WorldPresets;
+import net.minecraft.world.level.storage.LevelStorageSource;
+import net.minecraft.world.level.storage.PrimaryLevelData;
+import net.neoforged.neoforge.server.ServerLifecycleHooks;
+import org.apache.logging.log4j.LogManager;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.slf4j.Logger;
+
+/**
+ * A {@link ParameterResolver} that provides a {@link MinecraftServer} parameter.
+ * <p>
+ * The server is ephemeral, meaning that it doesn't store any data, and only has a void overworld available.
+ * <p>
+ * You should <strong>NOT</strong> not interact with the world of that server as it purely exists to load datapack data.
+ * If you need an actual world, you should use a {@linkplain net.minecraft.gametest.framework.GameTest GameTest} instead.
+ *
+ * <p>
+ * Example usage:
+ * 
+ * <pre>
+ * {@code
+ * @Test
+ * @ExtendWith(EphemeralTestServerProvider.class)
+ * void someJUnitTest(MinecraftServer server) {
+ *     assert server.registryAccess().registryOrThrow(Registries.ITEM).getTag(ItemTags.ANVIL).isPresent();
+ * }
+ * }
+ * </pre>
+ * 
+ * You can also annotate a class with {@link ExtendWith} to provide a server to all tests in that class.
+ *
+ * <p>
+ * The server instance is lazy (only created if a test needs it) and <strong>singleton</strong>.
+ * 
+ * @see ExtendWith
+ */
+public class EphemeralTestServerProvider implements ParameterResolver, Extension {
+    public static final AtomicReference<MinecraftServer> SERVER = new AtomicReference<>();
+    public static final AtomicBoolean IN_CONSTRUCTION = new AtomicBoolean();
+
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return parameterContext.getParameter().getType() == MinecraftServer.class;
+    }
+
+    @Override
+    public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return grabServer();
+    }
+
+    public static MinecraftServer grabServer() {
+        if (ServerLifecycleHooks.getCurrentServer() != null) {
+            return ServerLifecycleHooks.getCurrentServer();
+        }
+
+        if (!IN_CONSTRUCTION.get()) {
+            IN_CONSTRUCTION.set(true);
```

Comment:
This should use [`AtomicBoolean#compareAndSet`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/atomic/AtomicBoolean.html#compareAndSet(boolean,boolean)) instead.

```suggestion
        if (IN_CONSTRUCTION.compareAndSet(false, true)) {
```